src/base/rss/rss_session.cpp

```diff
@@ -140,7 +140,7 @@ nonstd::expected<void, QString> Session::addFolder(const QString &path)
     if (!result)
         return result.get_unexpected();
 
-    const auto destFolder = result.value();
+    auto *destFolder = result.value();
```

Comment:
Why `const` is removed here and in similar places?

Reply:
Previously, `const auto` is equal to `auto *const`. And since `*const` is a bit over cautious for short code, the `const` is dropped. Do you want it preserved?

Reply:
>BTW, in some other similar places `const` is added on the contrary.

So I was just confused by inconsistent changes.

Reply:
>Personally, I don't care about const in many cases (especially in short/trivial code). But if you promote its widespread use, then I don't see a fundamental difference between pointers and other types of data. Do you see?

Not sure I read you correctly, I only care about using `const T *` and not so much about `T *const` since the code of reassigning a pointer is unusual and quite easy to spot. And therefore I think having `const auto *` (with `*` explicitly written out) is better since it can enforce `const` on the pointed value and not on the address.

Reply:
>since the code of reassigning a pointer is unusual and quite easy to spot.

Is it any different from the code of reassigning an integer or something other? IMO, all those possible errors/typos that you are trying to avoid by declaring variables as `const` are equally applicable to any types, including pointers, aren't they? E.g.:
```c++
int a = getA();
// some code...
if (a = getB())
{
    // do something
}
```
and the same with pointers:
```c++
Class *a = getA();
// some code...
if (a = getB())
{
    // do something
}
```

At the same time, errors related to the declaration of `T *` rather than `const T *` seem less likely for me (unless we are not talking about function parameters).

Reply:
>IMO, all those possible errors/typos that you are trying to avoid by declaring variables as const are equally applicable to any types, including pointers, aren't they? 

I think there is a slight difference.
Broadly speaking, for a pointer, it is 'safe' to reassign pointer address or merely read its value, but not when you update the pointed value (as it is an irreversible change). For variables, updating a variable is always an uncertainty. Immutable variables are great for readability and therefore I would prefer minimize updates for a variable.

Reply:
>Broadly speaking, for a pointer, it is 'safe' to reassign pointer address or merely read its value, but not when you update the pointed value (as it is an irreversible change). 

Also, there is the case of another level of indirection, however I haven't been bitten to be worrying of it. (and hopefully I won't)